src/Models_Functions/QSM/calcChiL2.m

```diff
@@ -0,0 +1,61 @@
+function [chiL2, chiL2pcg] = calcChiL2(phaseUnwrapped, lambdaL2, directionFlag, imageResolution, mask, paddingSize, magnWeight)
+
+  N = size(phaseUnwrapped);
+
+  [fdx, fdy, fdz] = calcFdr(N, directionFlag);
+
+  FOV = N .* imageResolution;  % (in milimeters)
+
+  D = fftshift(kspaceKernel(FOV, N));
+
+  E2 = abs(fdx).^2 + abs(fdy).^2 + abs(fdz).^2;
+  D2 = abs(D).^2;
+
+
+  tic
+  D_reg = D ./ ( eps + D2 + lambdaL2 * E2 );
+  D_regx = ifftn(D_reg .* fftn(phaseUnwrapped));
+  toc
+
+
+  chiL2 = real(D_regx) .* mask;
+  chiL2 = chiL2(1+paddingSize(1):end-paddingSize(1),1+paddingSize(2):end-paddingSize(2),1+paddingSize(3):end-paddingSize(3));
+
+
+  % Memory cleanup in case magn weight will be performed
+  clear D_reg
+
+  %% L2-regularized solution with pre-conditionned magnitude weighting
+
+  if nargin>6 && exist('magnWeight', 'var') && nargout == 2
```

Comment:
@agahkarakuzu I would avoid using this function pattern (`nargin>6`) to check for optional variables. In my experience, it's very difficult to maintain in the long term, as for larger functions it may not be clear to another developper that this "magic number" (e.g. 6) exists. Bug can (and have) easily occur if another dev shifts around variables, or inserts new ones.